web/src/test/java/com/epam/rft/atsy/web/controllers/NewApplicationControllerTest.java

```diff
@@ -1,152 +1,235 @@
+    this.mockMvc.perform(get(REQUEST_URL_GET))
+        .andExpect(status().isOk())
+        .andExpect(view().name(VIEW_NAME))
+        .andExpect(forwardedUrl(VIEW_PREFIX + VIEW_NAME + VIEW_SUFFIX));
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeSuccessful() throws Exception {   
+    given(this.candidateService.getCandidate(1L)).willReturn(new CandidateDTO());
+    given(this.positionService.getPositionDtoById(1L)).willReturn(positionDTO);
+    given(this.channelService.getChannelDtoById(1L)).willReturn(channelDTO);
+
+    this.mockMvc.perform(post(REQUEST_URL_POST)
+        .param("candidateId", STRING_VALUE_ONE)
+        .param("position.id", STRING_VALUE_ONE).param("position.name", POSITION_NAME_DEVELOPER)
+        .param("channel.id", STRING_VALUE_ONE).param("channel.name", CHANNEL_NAME_FACEBOOK)
+        .param("description", DESCRIPTION))
+        .andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrl(REDIRECT_URL_FOR_CANDIDATE_A));
+
+    val applicationDTOCaptor = ArgumentCaptor.forClass(ApplicationDTO.class);
+    val stateHistoryDTOCaptor = ArgumentCaptor.forClass(StateHistoryDTO.class);
+
+    verify(applicationsService).saveApplication(applicationDTOCaptor.capture(), stateHistoryDTOCaptor.capture());
+    then(applicationsService).should()
+        .saveApplication(applicationDTOCaptor.getValue(), stateHistoryDTOCaptor.getValue());
+    then(this.candidateService).should().getCandidate(1L);
+    then(this.positionService).should().getPositionDtoById(1L);
+    then(this.channelService).should().getChannelDtoById(1L);
+
+    assertThat(stateHistoryDTOCaptor.getValue(), equalTo(stateHistoryDTO));
+    assertApplicationDtoWhenSaveOrUpdateIsSuccess(applicationDTOCaptor);
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeUnsuccessfulWhenCandidateIdIsNotANumber() throws Exception {
+
+    this.mockMvc.perform(post(REQUEST_URL_POST)
+        .param("candidateId", TEXT)
+        .param("position.id", STRING_VALUE_ONE).param("position.name", POSITION_NAME_DEVELOPER)
+        .param("channel.id", STRING_VALUE_ONE).param("channel.name", CHANNEL_NAME_FACEBOOK)
+        .param("description", DESCRIPTION))
+        .andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrl(REDIRECT_URL_FOR_NON_EXISTING_CANDIDATE));
+
+    verifyZeroInteractions(applicationsService);
+    then(this.candidateService).should().getCandidate(null);
+    then(this.positionService).should().getPositionDtoById(1L);
+    then(this.channelService).should().getChannelDtoById(1L);
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeUnsuccessfulWhenCandidateIdNotExists() throws Exception {
+    given(this.messageKeyResolver.resolveMessageOrDefault(COMMON_INVALID_INPUT_MESSAGE_KEY)).willReturn(COMMON_INVALID_INPUT_MESSAGE);
+    given(this.messageKeyResolver.resolveMessageOrDefault(CANDIDATE_NOT_FOUND_ERROR_MESSAGE_KEY)).willReturn(CANDIDATE_NOT_FOUND_ERROR_MESSAGE);
+

```

Comment:
There is no `given` initialization for the `candidateService`, `positionService` and the `channelService`.